### ISSUE TITLE
Confluence Auth using Personal Access Token not supported

### ISSUE THREAD

**Issue by yehoshuadimarsky** [sphinx-contrib/confluencebuilder#616]
### Problem
My org uses Confluence Enterprise, and I followed the instructions in their docs to create a Personal Access Token (here - https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html#UsingPersonalAccessTokens-UsingPATs). When I test it out using CURL as they suggest, it works fine, but it doesn't work with this library. 

### Cause
Digging into it I see that the reason is because it is supposed to be used in a header like this (in Python requests) `{'Authorization': 'Bearer <yourToken>'}`. 

### As a header
Per the [docs](https://sphinxcontrib-confluencebuilder.readthedocs.io/en/stable/configuration/#confval-confluence_publish_headers), I added these extra headers to my `conf.py` 
```python
confluence_publish_headers = {'Authorization': 'Bearer <yourToken>'}
```
https://github.com/sphinx-contrib/confluencebuilder/blob/e3f227e69bd83551546048a5212c0d50d4dc2e5c/sphinxcontrib/confluencebuilder/rest.py#L72-L74

That works - **if you skip the `auth`**. I tried it like this:
```python
import requests
base_url = 'REDACTED'
url = base_url + '/rest/api/content/123456789012'
session = requests.Session()
session.headers.update({
    'User-Agent': 'Sphinx Confluence Builder',
    'X-Atlassian-Token': 'no-check',
})
token = 'REDACTED'
confluence_publish_headers = {'Authorization': f'Bearer {token}'}
session.headers.update(confluence_publish_headers)
print(session.get(url).text)
```
### The `auth`
BUT - the problem is, you MUST provide the `auth`, either as a username and password, or as a custom class. Passing an empty dict works, but this library will discard it because it evaluates to `False`
https://github.com/sphinx-contrib/confluencebuilder/blob/e3f227e69bd83551546048a5212c0d50d4dc2e5c/sphinxcontrib/confluencebuilder/rest.py#L95-L101

### Summary
So, in summary, I cannot use the token in this library as it is intended to be used by Confluence.

**Comment by jdknight** [comment]
> ... you MUST provide the auth ...

I am a bit confused by this statement. Why is there a requirement for the `auth` attribute to be set? Or are you saying there is a situation where you want to tailor the `auth` entry but it does not work when needing to use PATs as well?

From the recent discussion in #611, yes, the manipulation of the [`confluence_publish_headers`](https://sphinxcontrib-confluencebuilder.readthedocs.io/configuration/#confval-confluence_publish_headers) configuration (or even using [`confluence_request_session_override`](https://sphinxcontrib-confluencebuilder.readthedocs.io/configuration/#confval-confluence_request_session_override)) is required to take advantage of PATs. Ideally, we will have a more graceful option to configure this in the next release. But I assume you may be talking about a different issue here?

**Comment by jdknight** [comment]
Note that there may be a chance to have a v1.8 made this upcoming weekend (tentative). If your interaction with a Confluence instance when using PATs can be resolved from [the changes outlined in the related issue](https://github.com/sphinx-contrib/confluencebuilder/issues/611#issuecomment-1046102268), great; however, if you are observing another issue or something else being overlooked, please let me know. This could allow additional corrections to be applied before the next release; otherwise it may be awhile before a following release is made.

**Comment by yehoshuadimarsky** [comment]
Thanks for all your work and comments, I appreciate it. Regarding

> I am a bit confused by this statement. Why is there a requirement for the `auth` attribute to be set? Or are you saying there is a situation where you want to tailor the `auth` entry but it does not work when needing to use PATs as well?

What I meant was that in this code,
https://github.com/sphinx-contrib/confluencebuilder/blob/e3f227e69bd83551546048a5212c0d50d4dc2e5c/sphinxcontrib/confluencebuilder/rest.py#L95-L101
It requires either `confluence_server_user` to be set (which we don't use when using a PAT), or a custom `confluence_server_auth` which also won't work.

**Comment by jdknight** [comment]
Ah ~ ok. I will assume this is a duplicate. Apologies though ~ the "proper" use of PATs were not really intuitive with the existing stable release (v1.7.1). Hopefully, with the recently added configuration and documentation changes available next release (v1.8), configuring to use PATs will much more explicit on how to use.

We can leave this issue opened (if desired), until the release is made. And if there are other concerns related to PATs which have been overlooked, please feel free to mention so.

**Comment by jdknight** [comment]
With the recently made v1.8 release (available on PyPI), `confluence_publish_token` should now provide a graceful way to configure for PATs. Marking as closed.
